sympy/matrices/tests/test_commonmatrix.py

```diff
@@ -510,6 +516,12 @@ def test_conjugate():
                           [ 5, 0]])
 
 
+def test_hermitian_conjugate_with_matrix_elements():
+    submat = Matrix([[0, I], [1, 0]])
+    mat = Matrix([[0, submat], [1, 0]])
+    assert mat.H == mat.adjoint()
```

Comment:
Since this test function is very similar to the one above I would merge them into one function. Also I would prefer the rhs of these asserts to be as explicit as possible.

You could have a bunch of
```
assert mat.H ==
assert mat.transpose() == 
assert mat.conjugate().transpose() ==
...
```
(same goes for matrix of operators)

These tests are cheap and I think it's good to show all possibilities right next to each other so it's clear to anyone looking at the test code exactly how/why they are similar/different and that any differences are intentional.